modules/internal/ChapelRange.chpl

```diff
@@ -406,14 +406,100 @@ module ChapelRange {
 
   pragma "last resort"
   proc chpl_compute_low_param_loop_bound(param low, param high) param {
-    compilerError("Range bounds must be integers of compatible types");
+    compilerError("Range bounds must be integers of compatible types in param for loop");
   }
 
   pragma "last resort"
   proc chpl_compute_low_param_loop_bound(low, high) {
     compilerError("param for loop must be defined over a bounded param range");
   }
 
+  proc chpl_compute_count_param_loop(param count: integral) param {
+    return count;
+  }
+
+  pragma "last resort"
+  proc chpl_compute_count_param_loop(param count) {
+    compilerError("ranges in param for loop can only have integral count");
+  }
+
+  pragma "last resort"
+  proc chpl_compute_count_param_loop(count) {
+    compilerError("ranges in param for loop must be integral literals");
+  }
```

Comment:
I would expect that the above two error overloads could be combined into one (non-param) case since param actuals can be passed to non-param formals.  This would also permit the error messages to be unified (where I'd imagine the unified message to be something like "in a param for-loop, the count operator requires a param integer value".

If we were to keep them separate, I'd refine the messages such that the second emphasizes that the problem is that the value is not a param (it need not be an integer literal, it could be an integer param) and the first to emphasize that the type is wrong (where perhaps the type received could be printed out as well).

For simplicity, I'd probably unify them, though I'd be fine with keeping them separate and refining them as well.

Reply:
Sir, I did this change.